src/pytorch_lightning/serve/servable_module_validator.py

```diff
@@ -0,0 +1,145 @@
+import time
+from multiprocessing import Process
+from typing import Any, Dict, Optional
+
+import requests
+import torch
+
+import pytorch_lightning as pl
+from pytorch_lightning.callbacks import Callback
+from pytorch_lightning.serve.servable_module import ServableModule
+from pytorch_lightning.strategies import DDPFullyShardedNativeStrategy, DDPFullyShardedStrategy, DeepSpeedStrategy
+from pytorch_lightning.utilities.imports import _RequirementAvailable
+from pytorch_lightning.utilities.model_helpers import is_overridden
+from pytorch_lightning.utilities.rank_zero import rank_zero_only
+
+_NOT_SUPPORTED_STRATEGIES = (
+    DeepSpeedStrategy,
+    DDPFullyShardedNativeStrategy,
+    DDPFullyShardedStrategy,
+)
+
+
+class ServableModuleValidator(Callback):
+
+    """The ServableModuleValidator callback enables to validate a model is servable following the ServableModule
+    API.
+
+    Arguments:
+        optimization: The format in which the model should be tested while being served.
+        server: The library used to evaluate the model serving. Default is FastAPI.
+        host: The host address associated the server.
+        port: The port associated the server.
+    """
+
+    def __init__(
+        self,
+        optimization: Optional[str] = None,
+        server: str = "fastapi",
+        host: str = "127.0.0.1",
+        port: int = 8080,
+    ):
+        super().__init__()
+        if not _RequirementAvailable("fastapi"):
+            raise ModuleNotFoundError("The package fastapi is required by the ServableModuleValidator callback.")
+        if not _RequirementAvailable("uvicorn"):
+            raise ModuleNotFoundError("The package uvicorn is required by the ServableModuleValidator callback.")
+
+        # TODO: Add support for those optimizations
+        assert optimization in (None, "trace", "script", "onnx", "tensor_rt")
+        if optimization is not None:
+            raise NotImplementedError(f"The optimization {optimization} is currently not supported.")
+
+        # TODO: Add support for testing with those server services
+        assert server in ("fastapi", "ml_server", "torch_serve", "sagemaker")
```

Comment:
You could remove this assertion and type server as

```python
from typing_extensions import Literal

server: Literal["fastapi"]
```

then IDEs and type checkers will do the job of complaining and you can remove this assert

Reply:
This is meant primarily for the readers and to show the ambition to support more.